UnitTests/OIDTVAuthorizationRequestTests.m

```diff
@@ -0,0 +1,198 @@
+
+/*! @brief Expected HTTP Method for the authorization URLRequest
+ */
+static NSString *const kHTTPPost = @"POST";
+
+/*! @brief Expected ContentType header key for the authorization URLRequest
+ */
+static NSString *const kHTTPContentTypeHeaderKey = @"Content-Type";
+
+/*! @brief Expected ContentType header key for the authorization URLRequest
+ */
+static NSString *const kHTTPContentTypeHeaderValue =
+    @"application/x-www-form-urlencoded; charset=UTF-8";
+
+@implementation OIDTVAuthorizationRequestTests
+
+- (OIDTVServiceConfiguration *)testServiceConfiguration {
+  NSURL *tokenEndpoint = [NSURL URLWithString:kInitializerTestTokenEndpoint];
+  NSURL *TVAuthorizationEndpoint = [NSURL URLWithString:kInitializerTestTVAuthEndpoint];
+
+  OIDTVServiceConfiguration *configuration =
+      [[OIDTVServiceConfiguration alloc] initWithTVAuthorizationEndpoint:TVAuthorizationEndpoint
+                                                           tokenEndpoint:tokenEndpoint];
+  return configuration;
+}
+
+- (NSDictionary<NSString *, NSString *> *)bodyParametersFromURLRequest:(NSURLRequest *)urlRequest {
+  NSString *bodyString = [[NSString alloc] initWithData:urlRequest.HTTPBody
+                                               encoding:NSUTF8StringEncoding];
+  NSArray *bodyParameterStrings = [bodyString componentsSeparatedByString:@"&"];
+  
+
+  NSMutableDictionary<NSString *, NSString *> *bodyParameters = [[NSMutableDictionary alloc] init];
+
+  for (NSString *paramString in bodyParameterStrings) {
+    NSArray *components = [paramString componentsSeparatedByString:@"="];
+    NSLog(@"%@:%@", components[0], components[1]);
+    
+    [bodyParameters setObject:components[1] forKey:components[0]];
+  }
+
+  return bodyParameters;
+}
+
+- (void)testInitializer {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  XCTAssertEqualObjects(authRequest.responseType, OIDResponseTypeCode);
+  XCTAssertEqualObjects(authRequest.redirectURL, [[NSURL alloc] init]);
+}
+
+- (void)testURLRequestBasicClientAuth {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  NSURLRequest *urlRequest = [authRequest URLRequest];
+
+  XCTAssertEqualObjects([urlRequest HTTPMethod], kHTTPPost);
```

Comment:
Use dot notation for accessing properties (urlRequest.HTTPMethod).

https://groups.google.com/a/google.com/g/ios-apps-all/c/FancbJhO-k4/m/5RsOkGIpCAAJ